src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -656,8 +657,14 @@ private BoundExpression BindTupleExpression(TupleExpressionSyntax node, Diagnost
                 // validate name if we have one
                 if (name != null)
                 {
-                    // PROTOTYPE: check for a case "ItemX" where X is not i
+                    elementNameCount++;
 
+                    if (TupleTypeSymbol.IsMemberNameReserved(name) && name != TupleTypeSymbol.UnderlyingMemberName(i))
```

Comment:
Correct. Much doesn't work above tuple-7. 
I'll add a `NotImplementedException("PROTOTYPE")` for this case in `UnderlyingMemberName`.